buildingBlocks.js

```diff
@@ -516,9 +515,12 @@ class buildingBlocks extends Phaser.Scene {
             this.room2_E_KeyImg.y = this.room2_character_north.y-75;
             this.room2_E_KeyImg.alpha = 1.0;
             if (this.room2_key_E.isDown) {
+                
                 if(roomProgress <= 2000)
+                    this.coin0.alpha = 1.0;
                     roomProgress = 2005;
```

Comment:
Need { braces } to make a compound statement containing these 2 assignments for the "then" part...

Reply:
All assignments of roomProgress need to be "guarded" by an if-condition. If we went further (e.g., roomProgress became 4000) then come back here and hit that upper middle panel then our progress falls backwards to 2005.